applications/MeshingApplication/custom_processes/metric_fast_init_process.h

```diff
@@ -0,0 +1,278 @@
+// KRATOS  __  __ _____ ____  _   _ ___ _   _  ____ 
+//        |  \/  | ____/ ___|| | | |_ _| \ | |/ ___|
+//        | |\/| |  _| \___ \| |_| || ||  \| | |  _ 
+//        | |  | | |___ ___) |  _  || || |\  | |_| |
+//        |_|  |_|_____|____/|_| |_|___|_| \_|\____| APPLICATION
+//
+//  License:		 BSD License
+//                       license: MeshingApplication/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrándiz
+//
+
+#if !defined(KRATOS_METRIC_FAST_INIT_PROCESS)
+#define KRATOS_METRIC_FAST_INIT_PROCESS
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "processes/process.h"
+#include "meshing_application.h"
+#include "includes/model_part.h"
+
+namespace Kratos
+{
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+    
+///@}
+///@name  Enum's
+///@{
+    
+///@}
+///@name  Functions
+///@{
+    
+/// Short class definition.
+// This process initializes the variables related with the ALM 
+/** Detail class definition.
+*/
+template<unsigned int TDim>  
+class MetricFastInit
+    : public Process
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of MetricFastInit
+    KRATOS_CLASS_POINTER_DEFINITION(MetricFastInit);
+    
+    // General type definitions
+    typedef Node<3>                                          NodeType;
+    typedef Geometry<NodeType>                           GeometryType;
+    typedef ModelPart::NodesContainerType              NodesArrayType;
+    typedef ModelPart::ConditionsContainerType    ConditionsArrayType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Default constructor.
+    MetricFastInit( ModelPart& rThisModelPart):mrThisModelPart(rThisModelPart)
+    {
+        KRATOS_TRY;
+        
+        KRATOS_CATCH(""); 
+    }
+
+    /// Destructor.
+    virtual ~MetricFastInit()
+    {
+    }
+
+    ///@}
+    ///@name Access
+    ///@{
+
+    ///@}
+    ///@name Inquiry
+    ///@{
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    ///@}
+    ///@name Friends
+    ///@{
+    
+    ///@}
+    ///@name Operators
+    ///@{
+
+    void operator()()
+    {
+        Execute();
+    }
+    
+    ///@}
+    ///@name Operations
+    ///@{
+    
+    virtual void Execute() override
+    {
+        KRATOS_TRY;
+        
+        constexpr unsigned int size = TDim == 2  ? 3: 6;
+        
+        const array_1d<double, size> zerovector(size, 0.0);
+        
+        // We iterate over the node
+        NodesArrayType& pNodes = mrThisModelPart.Nodes();
+        auto numNodes = pNodes.end() - pNodes.begin();
+        
+//         #pragma omp parallel for 
```

Comment:
this loop is surely parallel. recall to define your zerovector as "firstprivate"

also do

      it_node->SetValue(MMG_VECTOR,zerovector)